webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpParallelCoordinates.tsx

```diff
@@ -85,69 +84,79 @@ const HpParallelCoordinates: React.FC<Props> = ({
     return undefined;
   }, [ experiment.config.searcher, selectedMetric ]);
 
+  const handleFilterChange = useCallback((hermesFilters: Hermes.Filters) => {
+    // Skip if there aren't any chart data.
+    if (!chartData) return;
+
+    // Initialize a new trial id filter map.
+    const newFilteredTrialIdMap = chartData.trialIds.reduce((acc, trialId) => {
```

Comment:
`question` `non-blocking` could we memoize this based on `chartData` somehow?

Reply:
it's possible but it would require storing the `hermesFilters` as a state, and then setting up the filter id map to be memoized based on both the `chartData.trialIds` and `hermesFilters`. This setup has the advantage of avoiding having to save an extra state, which actually has a good chance of avoiding extra state update cycles.
```
handleFilterChange => setHermesFilters => hermesFilters updates => filteredTrialIdMap memo updates
vs just
handleFilterChange => setFilteredTrialIdMap
```